### PR TITLE
Fix string to text migrations for Oracle

### PR DESCRIPTION
**Pull request by aidenfoxx**
Some Oracle migrations fail when converting between text and string types with `Error: alter table "directus_webhooks" modify "url" clob - ORA-22858: invalid alteration of datatype`.

This obviously isn't a perfect fix, but I think it's better than not being able to install Directus, and seems to have minimal impact.

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
> This obviously isn't a perfect fix, but I think it's better than not being able to install Directus, and seems to have minimal impact.

**Comment by aidenfoxx** [comment]
Yeah, I know how it is. And more often than not temporary fixes often become permanent fixes. However, while these migrations exist they do block Oracle installations.

If I have time I might get around to trying to solve the issue the correct way in Knex, but I'm not sure when that will be.

**Comment by rijkvanzanten** [comment]
We haven't heard back on the Knex side of things, but in the meantime, it might work if we update the migration to be something like (pseudo code):

```js
async function up(knex) {
	await knex.schema.alterTable(table => {
		table.text('temp_text_collections');
	});

	await knex('directus_webhooks').update({ temp_text_collections: knex.ref('collections') });

	await knex.schema.alterTable(table => {
		table.dropColumn('collections');
		table.renameColumn('temp_text_collections', 'collections');
	});
}
```

?

**Comment by aidenfoxx** [comment]
Seems reasonable. I can look into implementing it tomorrow.

**Comment by aidenfoxx** [comment]
This is made incredibly difficult with how Knex generates alter queries. It seems you always have to specify the column type which generates queries like:

```
/* ERROR: ORA-22296: invalid ALTER TABLE option for conversion of LONG datatype to LOB */
alter table "directus_webhooks" modify "collections_temp" clob not null
```

Unfortunately Oracle throws an error for this. However, if we don't specify the column type it's fine:

```
alter table "directus_webhooks" modify "collections_temp" not null
```

I can't think of a nice way around this. We have to generate the new column as nullable, or define a default value. But the same issue occurs if you try and remove a default value.

File: api/src/utils/oracle-schema.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Knex } from 'knex';
+
+export async function oracleForceAlterColumn(knex: Knex, table: string, column: string, type: string): Promise<void> {
+	await knex.raw(`ALTER TABLE "${table}" ADD "${column}__temp" ${type}`);
```
Comment (rijkvanzanten) [inline api/src/utils/oracle-schema.ts:10]: Lets use Knex's identifier syntax here to make sure we have some sql injection safety here (even though it's only us that uses this now):

```suggestion
	await knex.raw(`ALTER TABLE ?? ADD ?? ?`, [table, `${column}__temp`, type]);
```

**Review by rijkvanzanten — APPROVED** [review]

**Comment by sflepp** [comment]
Hello everybody, is this problem fixed in v9.0.0-rc.74?

I'm still getting following an error when I run 

```
npx create-directus-project project-name
````

```
✨ Applying Change Webhook URL Type...

Something went wrong while seeding the database:

[Error] alter table "directus_webhooks" modify "url" clob - ORA-22858: invalid alteration of datatype

Please try again

```

Thanks for responding.

**Comment by aidenfoxx** [comment]
So a workaround was added, but to ensure the workaround didn't affect Oracle we had it check the env for the correct `DB_CLIENT`. It seems this variable never gets set during the initialization. A workaround for now would be to run:

```
DB_CLIENT=oracledb npx create-directus-project project-name
```

Or for windows:
```
set DB_CLIENT=oracledb&&npx create-directus-project project-name
```

**Comment by aidenfoxx** [comment]
@sflepp This should be properly fixed with #6228.

**Pull request merged**